module/os/freebsd/zfs/zfs_debug.c

```diff
@@ -29,14 +29,14 @@
 typedef struct zfs_dbgmsg {
 	list_node_t zdm_node;
 	time_t zdm_timestamp;
-	int zdm_size;
+	uint32_t zdm_size;
 	char zdm_msg[1]; /* variable length allocation */
 } zfs_dbgmsg_t;
 
 static list_t zfs_dbgmsgs;
-static int zfs_dbgmsg_size = 0;
+static uint32_t zfs_dbgmsg_size = 0;
 static kmutex_t zfs_dbgmsgs_lock;
-int zfs_dbgmsg_maxsize = 4<<20; /* 4MB */
+uint32_t zfs_dbgmsg_maxsize = 4<<20; /* 4MB */
```

Comment:
As nice as it would be to use a fixed size type here, the Linux kernel module interfaces expect either an `int`, `unsigned int`, `long` or `unsigned long` type.  Internally, we can cast these values as we need too, but since we're registering them with the kernel it seemed prudent to stick to the expected types to avoid any unexpected behavior across architectures.

Reply:
`unsigned int` is the same as `uint32_t` across all architectures. `cstyle.pl` does not permit using uint and most of the code already uses `uint32_t`, so I used `uint32_t` to be consistent.

That said, the same nice consistency is not true for `long`, which is why I did not make any module parameters use `uint64_t`.

Reply:
> unsigned int is the same as uint32_t across all architectures.

Well, it's true for architectures which use the ILP32, LP64, or LLP64 data model, and realistically we only really support LP64 with OpenZFS (ILP32 builds but has some known issues).  It's not true for LP32, but that's ancient and not a concern.  So, I think you're points fair we this should be fine.